service/controller/clusterapi/v29/resource/tcnp/create.go

```diff
@@ -29,6 +29,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if len(cc.Spec.TenantCluster.TCNP.AvailabilityZones) == 0 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cannot ensure the tenant cluster's node pool cloud formation stack")
```

Comment:
I think this line is not useful if there is not corresponding `ensuring ...` (then also it would become `did not ensure`).